service/telemetry/config.go

```diff
@@ -137,9 +127,45 @@ type TracesConfig struct {
 func (c *Config) Validate() error {
 
 	// Check when service telemetry metric level is not none, the metrics address should not be empty
-	if c.Metrics.Level != configtelemetry.LevelNone && c.Metrics.Address == "" {
+	if c.Metrics.Level != configtelemetry.LevelNone && c.Metrics.Address == "" && len(c.Metrics.Readers) == 0 {
 		return fmt.Errorf("collector telemetry metric address should exist when metric level is not none")
 	}
 
 	return nil
 }
+
+func (mrs *MeterProviderJsonMetricReaders) Unmarshal(conf *confmap.Conf) error {
+	if err := conf.Unmarshal(mrs); err != nil {
+		return err
+	}
+	if *mrs == nil {
```

Comment:
Should this be `if mrs == nil`?

Reply:
Also, can we add a test case that can hit this? Is it even possible for `conf.Unmarshal` to not return an error but `mrs` be nil?

Reply:
I think you're right, this check doesn't make sense here. Looking through the rest of the code it does look like there should be a check for `conf == nil` before line 138 though.